### DISCUSSION TITLE
Can shell calls be made in config files?

### DISCUSSION THREAD

**Discussion by babudro** [fail2ban/fail2ban#3296] [Q&A] [answered]
I am trying to develop a master set of fail2ban config files that I can share among all my servers.  I created jail.local files, abuseIPDB key files, and path override files for each server that are named according to the host name.  My current method is to symlink the proper file on each server, e.g.:

`ln -s paths/server-name paths-overrides.local`

I would like to find a way to grab the short host name inside a config file so I can skip the symlink and do something like this:

`after = paths/$(hostname -s)`

I tried using `after = paths/<fq-hostname>`, `after = paths/<HOST>`, and `after = paths/%(hostname)s`.  Using a variable like `<var>` silently fails; using Python notation like `%(hostname)s` throws an error.

Is there any way this can be done?

**Comment by sebres** [comment, marked as answer]
> Is there any way this can be done?

No. There is no hostname interpolation at the includes stage at the moment. 

It would expect an enhancement of the config reader, for instance somewhere here:
https://github.com/fail2ban/fail2ban/blob/5bfd9992b470b724463ace3b807bf9cc43bea42b/fail2ban/client/jailreader.py#L127
```diff
		self.merge_defaults({
+			"fq_hostname":	str(DNSUtils.getHostname(fqdn=True)),
+			"sh_hostname":	str(DNSUtils.getHostname(fqdn=False)),
```
then you'd be able to use `%(fq_hostname)s` or `%(sh_hostname)s` (I don't think minus `-` char is allowed within pythons substitution).
`<fq-hostname>` and `<sh-hostname>` are dynamic interpolations and only working in actions, not to mention that such a substitution will be probably impossible at the include stage without some special handling.

**Reply by babudro** [reply]
OK, thank you for the explanation.
